### PR TITLE
Add release note for gfs.v16.3.14 - adding wdqms into global-workflow

### LINKED ISSUE #2389: GFSv16.3.16 - Add WDQMS processing job
### Description

WDQMS (WIGOS Data Quality Monitoring System) is a tool generating data quality reports for conventional and marine data.
It was developed by DA team and is being run in real time by Obsproc.
Once a day, the code uses the four gdas.YYYYMMDD/HH/atmos/gdas.tHHz.cnvstat files from the prior day to generate 12 reports:
NCEP_TEMP_YYYYMMDD_HH.csv
NCEP_SYNOP_YYYYMMDD_HH.csv
NCEP_MARINE_YYYYMMDD_HH.csv
where HH=00,06,12,18

The reports are then staged on emcrzdm ftp for downstream users:
https://ftp.emc.ncep.noaa.gov/wdqms/ncep

Code: https://github.com/kevindougherty-noaa/wdqms

Currently, the cron schedule for WDQMS is:
```bash
07 14 * * * bash /lfs/h2/emc/obsproc/noscrub/ashley.stanfield/wdqms_new/scripts/WDQMS_new.sh
```
and a run lasts about 15 minutes on a dev machine.

The operationalization of this tool will prevent reporting gaps and delays during prod switches, wcoss tests, etc.

Requirements
Generate 12 monitoring reports per day and stage on emcrzdm ftp, in a prescribed directory

Acceptance Criteria (Definition of Done)
The 12 reports are generated and staged by 12 UTC of the next day

** Suggest A Solution**
We propose to include WDMQS as a step in the $pslot.xml file

Subsumes #1459

### Target version

v16.3.16

### Tasks

- [x] Create release branch - [release/gfs.v16.3_wdqms](https://github.com/NOAA-EMC/global-workflow/tree/release/gfs.v16.3_wdqms)
- [x] Make workflow changes for upgrade in release branch (See PR #2396)
- [x] Create release notes
- [x] Cut hand-off tag for CDF
- [x] Submit CDF to NCO
- [x] Absorb updates from NCO.
- [x] Recut hand-off tag.
- [x] Resubmit CDF (?)
- [x] Implementation into operations complete
- [x] Merge release branch into operational branch
- [x] Cut version tag from operational branch
- [x] Release new version tag
- [x] Announce to users
- [ ] Update Read-The-Docs operations status version in develop

### PR DESCRIPTION
**Pull request by emilyhcliu**
# Description

# Type of change

- Bug fix (fixes something broken)
- New feature (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES/NO
- Does this change require a documentation update? YES/NO

# How has this been tested?

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: docs/Release_Notes.gfs.v16.3.15.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+SORC CHANGES
+------------
+
+* No changes from GFS v16.3.13
+
+JOBS CHANGES
+------------
+
+* No changes from GFS v16.3.13
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.15.md]:
```suggestion
* Added a new `j-job` `JGDAS_ATMOS_ANALYSIS_WDQMS`, corresponding `exscript` `exgdas_atmos_analysis_wdqms.sh` and a utility in `ush` `wdqms.py`.
* Added `ecf` script `jgdas_atmos_analysis_wdqms.ecf` and updated the `ecflow` suite definition files.
```

File: docs/Release_Notes.gfs.v16.3.15.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+PARM/CONFIG CHANGES
+-------------------
+
+* No changes from GFS v16.3.13
+
+SCRIPT CHANGES
+--------------
+
+* No changes from GFS v16.3.13
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.15.md]:
```suggestion
* Added `exscript` `exgdas_atmos_analysis_wdqms.sh`
```

File: docs/Release_Notes.gfs.v16.3.15.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+FIX CHANGES
+-----------
+
+* No changes from GFS v16.3.13
+
+MODULE CHANGES
+--------------
+
+* No changes from GFS v16.3.13
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.15.md]:
```suggestion
* New job loads `python` module in the ecf script.
```

File: docs/Release_Notes.gfs.v16.3.15.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+JOBS CHANGES
+------------
+
+* No changes from GFS v16.3.13
+
+PARM/CONFIG CHANGES
+-------------------
+
+* No changes from GFS v16.3.13
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.15.md]:
```suggestion
* Added a `config.wdqms` file for the new j-job added.
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
I did my best to provide input on the changes in jobs/scripts etc.  Please feel free to update as required.

**Comment by KateFriedman-NOAA** [comment]
@emilyhcliu Two important things:

1. This PR should be modifying the [existing skeleton release notes I created in the release branch](https://github.com/NOAA-EMC/global-workflow/blob/release/gfs.v16.3_wdqms/docs/Release_Notes.gfs.v16.3.14.md) and not adding a new release notes.
2. This is tentatively v16.3.14 and should be noted as such until NCO tells us otherwise (I can handle the version change if this isn't going to be v16.3.14). I set the tentative version in the skeleton release notes, so you just need to add the technical/science details.

Please change this PR to send your updates into the existing [Release_Notes.gfs.v16.3.14.md](https://github.com/NOAA-EMC/global-workflow/blob/release/gfs.v16.3_wdqms/docs/Release_Notes.gfs.v16.3.14.md) or reinitiate this PR to do that. Thanks!

**Comment by emilyhcliu** [comment]
@aerorahul @KateFriedman-NOAA 
I have not started yet. I am simply creating a new release note by copying the previous one. So, do not worry about the content now. I will update the release note tomorrow.  
This is to prepare to work on the release note tomorrow.

File: docs/Release_Notes.gfs.v16.3.14.md [hunk 5]
```diff
@@ -60,17 +60,18 @@ SORC CHANGES
 JOBS CHANGES
 ------------
 
-* No changes from GFS v16.3.13
+* Added a new `j-job` `JGDAS_ATMOS_ANALYSIS_WDQMS`, corresponding `exscript` `exgdas_atmos_analysis_wdqms.sh` and a utility in `ush` `wdqms.py`.
+* Added `ecf` script `jgdas_atmos_analysis_wdqms.ecf` and updated the `ecflow` suite definition files.
```
Comment (KateFriedman-NOAA) [inline docs/Release_Notes.gfs.v16.3.14.md]: So the "JOBS CHANGES" section should be information about the new job added to the suite. So here, add mention of the job being added to the system and very briefly what it does.

Please move the other info about new scripts down to the "SCRIPT CHANGES" section.

**Review by KateFriedman-NOAA — CHANGES_REQUESTED** [review]
Thanks for fixing the release notes filename and version.

I left an inline comment to address but also need the following done in the release notes:

1. Update the PRELUDE section with a brief description of this upgrade. This is effectively the title of the upgrade.
2. Update the "JOB DEPENDENCIES AND FLOW DIAGRAM" section to add mention of the job dependency for the new wdqms job (what job needs to be complete before this new job runs).
3. @emilyhcliu add your name after my name in the "PREPARED BY" section at the very bottom.

Thanks! :)

File: docs/Release_Notes.gfs.v16.3.14.md [hunk 6]
```diff
@@ -60,17 +60,24 @@ SORC CHANGES
...
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.14.md]:
```suggestion
* Added a new `j-job` `JGDAS_ATMOS_ANALYSIS_WDQMS`
```

File: docs/Release_Notes.gfs.v16.3.14.md [hunk 7]
```diff
@@ -60,17 +60,24 @@ SORC CHANGES
...
+  - `config.base.nco.static`
+* Added resource parameters for `WDQMS` job in the following resource configuration files
+  - `config.resource.emc.dyn`
+  - `config.resource.nco.static`
 
 SCRIPT CHANGES
 --------------
 
-* No changes from GFS v16.3.13
+* Added `exscript` `exgdas_atmos_analysis_wdqms.sh`
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.14.md:81]:
```suggestion
* Added `exscript` `exgdas_atmos_analysis_wdqms.sh`
* Added a utility in `ush` `wdqms.py` called by the new job.
* Added `ecf` script `jgdas_atmos_analysis_wdqms.ecf` and updated the `ecflow` suite definition files.
```

**Review by aerorahul — COMMENTED** [review]
Added code change suggestions based on review.

File: docs/Release_Notes.gfs.v16.3.14.md (hunk 5 above)
Comment (aerorahul) [reply docs/Release_Notes.gfs.v16.3.14.md]: applied.  Please review if acceptable.

Comment (KateFriedman-NOAA) [reply docs/Release_Notes.gfs.v16.3.14.md]: Scripts changes section looks good now, thanks!

**Comment by emilyhcliu** [comment]
@KateFriedman-NOAA @aerorahul The release note is ready for final review. Please let me know if there is something I can improve or add.  Thanks!

The information not included is the `Dissemination Information`. The decision on this section is pending with the NCO.

File: docs/Release_Notes.gfs.v16.3.14.md [hunk 8]
```diff
@@ -4,7 +4,9 @@ GFS V16.3.14 RELEASE NOTES
 PRELUDE
 -------
 
-TODO: Fill in prelude short description of change
+This upgrade adds a new GDAS job - `gdaswdqms`. It generates 6-hourly summaries of observation reports, which contain information on time, location, station ID, and quality (usage flag and innovations) from GDAS analysis. The objective of this new job is to support the WMO WDQMS (WIGOS Data Quality Monitoring System) project and ensure that MWO observational data and products are reliable and correspond to agreed-upon needs.
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.14.md]:
```suggestion
This upgrade adds a new GDAS job - `gdaswdqms`. It generates 6-hourly summaries of observation reports, which contain information on time, location, station ID, and quality (usage flag and innovations) from GDAS analysis. The objective of this new job is to support the WMO WDQMS (WIGOS Data Quality Monitoring System) project and ensure that WMO observational data and products are reliable and correspond to agreed-upon needs.  
```

File: docs/Release_Notes.gfs.v16.3.14.md [hunk 9]
```diff
@@ -60,17 +62,25 @@ SORC CHANGES
...
-* No changes from GFS v16.3.13
+* Added a new `j-job` `JGDAS_ATMOS_ANALYSIS_WDQMS`
 
 PARM/CONFIG CHANGES
 -------------------
 
-* No changes from GFS v16.3.13
+* Added a `config.wdqms` file for the new j-job added.
+* Added new configuration parameter `DO_WDQMS` and set it to `YES` in the following configuration files
+  - `config.base.emc.dyn`
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.14.md]:
```suggestion
```
I think it is a `NO` in `config.base.emc.dyn`.
It should be noted that `config.base.emc.dyn` was also updated.

File: docs/Release_Notes.gfs.v16.3.14.md [hunk 10]
```diff
@@ -80,17 +90,29 @@ FIX CHANGES
...
-* No changes from GFS v16.3.13
+* Computing resource added for the `gdaswdqms` job for each gdas cycle: 
+
+  ```
+    export wtime_wdqms="00:20:00"
+    export npe_wdqms=1
+    export nth_wdqms=1
+    export npe_node_wdqms=1
+    export memory_wdqms="48GB"
+  ```
```
Comment (aerorahul) [inline docs/Release_Notes.gfs.v16.3.14.md]:
```suggestion
The new job `gdaswdqms` requests a single (1) core with `48GB` memory for 20 minutes
```

File: docs/Release_Notes.gfs.v16.3.14.md (hunk 9 above)
Comment (emilyhcliu) [reply docs/Release_Notes.gfs.v16.3.14.md]: True!!  I will modify it.
Thanks!

**Review by KateFriedman-NOAA — APPROVED** [review]
Updates for release notes look good. Thanks @emilyhcliu and @aerorahul for the updates. Approving and will merge into release branch.

**Pull request merged**